### ISSUE TITLE
Gradle plugin should contribute a parent, a dependency management import, or individual versions to generated poms

### ISSUE THREAD

**Issue by StephenDillon** [spring-projects/spring-boot#1716]
Not sure if this falls as a gradle maven plugin issue or spring boot issue.

When you have spring boot's blessed dependencies in your gradle build file, and use the gradle maven plugin to push a spring boot WAR application to artifactory, the generated pom file is invalid and rejected by the artifactory. 
I tracked this down to the generated pom file not having the spring blessed version's. (ie. pom has group and name but no version)

(GRADLE-2907 is open to improve the reporting of this issue by gradle, but the core issue could still be fixed)

The project i have is a internal EMC project but i can provide versions if required (using boot 1.1.5.RELEASE and gradle 2.1) or test any fixes.

The workaround is to add all version's and not use the blessed versioning.

**Comment by rburgst** [comment]
Hi. I think we are on the wrong track here. Actually from a pom perspective if you upload the fat jar, then the file should not have ANY dependencies since they are all packaged in. 
Am I holding it wrong?

What I try is to upload our final applications / services into artifactory before publishing (building docker containers to be precise) from there. However, if I use gradle dependencies to download the jars and build the docker container then I get the same problem with the invalid pom.

However, in such a case (where fat jars are uploaded to artifactory), it should not pull any other dependencies since they are packaged into the fat jar. 

so as far as I am concerned the generated pom should not have *any* dependencies.

Any suggestions?

**Comment by StephenDillon** [comment]
True the fat jar/war would have all the dependencies with it, but the generated pom is still useful to see included dependency versions and in my case Artifactory validated the pom on upload, so it rejected the upload. Adding the version or the parent should give a good pom thought.

**Comment by rburgst** [comment]
Well I guess what I would need is at least a different artefact type or something like that. Does anyone know how to influence the generated pom so that it doesn't have dependencies?

**Comment by cfirestone** [comment]
@wilkinsona When I try the workaround of setting the parent project for the pom generation, it does attempt to download the dependencies for the spring-boot-start-parent project, but it seems to only want to grab them from maven central (http://repo1.maven.org/maven2). 
```
:uploadArchives (Thread[main,5,main]) started.
:uploadArchives
Executing task ':uploadArchives' (up-to-date check took 0.0 secs) due to:
  Task has not declared any outputs.
Publishing configuration: configuration ':archives'
Publishing to org.gradle.api.publication.maven.internal.ant.DefaultGroovyMavenDeployer@276c179d
Downloading: org/springframework/boot/spring-boot-starter-parent/1.2.2.RELEASE/spring-boot-starter-parent-1.2.2.RELEASE.pom from repository central at http://repo1.maven.org/maven2
```
Is there any way for it to be configured to grab the dependencies from a private maven repository, like the one I have set for my app's compile dependencies?

**Comment by wilkinsona** [comment]
Unfortunately not. The use of Maven Central is hardcoded in Gradle or the Ant task that Gradle's reusing.

**Comment by wilkinsona** [comment]
This has been resolved in Boot 1.3 by the introduction of the dependency management plugin
